tests/dummy/app/templates/docs/glimmer-components.md

```diff
@@ -0,0 +1,109 @@
+# Glimmer Components
+
+<aside>
+
+If you’re entirely unfamiliar with Glimmer components, we recommend [the Ember Guides][guides-component] to understand that first!
+
+</aside>
+
+
+Glimmer Components are defined in one of three ways: with templates only, with a template and a backing class, or with only a backing class (i.e. a `yield`-only component). When using a backing class, you get a first-class experience using TypeScript! Unfortunately, we don’t yet support type-checking for templates, but we hope to build that out eventually. Don’t let that stop you, though: types in your component classes make for a great experience, so let’s dig in and see how it works in practice.
+
+## A simple component
+
+A *very* simple Glimmer component which lets you change the count of a value might look like this:
+
+```htmlbars
+<button {{on "click" this.minus}}>&minus;</button>
+{{this.count}}
+<button {{on "click" this.plus}}>+</button>
+```
+
+```ts
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking'
+import { action } from '@ember/object';
+
+export default class Counter extends Component {
+  @tracked count = 0;
+  
+  @action
+  plus() {
+    this.count += 1;
+  }
+  
+  @action
+  minus() {
+    this.count -= 1;
+  }
+}
+```
+
+Notice that there are no type declarations here – but this *is* actually a well-typed component. The type of `count` is `number`, and if we accidentally wrote something like `this.count = null` the compiler would give us an error.
+
+## A component with arguments
+
+So far so good, but of course most components aren’t quite this simple! Instead, they’re invoked by other templates and they can invoke other components themselves in their own templates.
+
+Glimmer components can receive both *arguments* and *attributes* when they are invoked. When you are working with a component’s backing class, you have access to the arguments but *not* to the attributes. The arguments are passed to the constructor, and then available as `this.args` on the component instance afterward. Let’s imagine a component which just logs the names of its arguments when it is first constructed:
+
+```ts
+import Component from '@glimmer/component';
+
+export default class ArgsDisplay extends Component {
+  constructor(owner: unknown, args: {}) {
+    super(owner, args);
+
+    Object.keys(args).forEach(argName => {
+      console.log(argName);
+    });
+  }
+}
+```
+
+<aside>
+
+If you’re used to the classic Ember Object model, there are two important differences in the constructor itself:
+
+- we use `super` instead of `this._super`
+- we *must* call `super` before we do anything else
```

Comment:
it might be useful to link to the part of the ECMA spec that states this as a hard requirement (I believe it is). 

This is the closest I could find
https://tc39.es/ecma262/#sec-runtime-semantics-classdefinitionevaluation

Reply:
Excellent suggestion, will do.

Reply:
Curiously, that doesn't seem to actually specify it. All the implementations throw a reference error, treating `this` as an uninitialized variable—i.e. it's a Temporal Dead Zone—but I can't find *any* reference to this in spec or other official docs.

Reply:
Added a note in b7e4e99; I don't love it but it's better than nothing.

Reply:
Resolved by 42fe7566.